samza-sql/src/main/java/org/apache/samza/sql/planner/SamzaSqlValidator.java

```diff
@@ -0,0 +1,279 @@
+    RelDataType projectFieldType;
+
+    // JavaTypes are relevant for Udf argument and return types
+    // TODO: Support UDF argument validation. Currently, only return types are validated and argument types are
+    //  validated during run-time.
+    if (selectQueryFieldType instanceof RelDataTypeFactoryImpl.JavaType) {
+      projectFieldType = new SamzaSqlJavaTypeFactoryImpl().toSql(selectQueryFieldType);
+    } else {
+      projectFieldType = selectQueryFieldType;
+    }
+
+    SqlTypeName outputSqlType = outputFieldType.getSqlTypeName();
+    SqlTypeName projectSqlType = projectFieldType.getSqlTypeName();
+
+    if (projectSqlType == SqlTypeName.ANY || outputSqlType == SqlTypeName.ANY) {
+      return true;
+    } else if (outputSqlType != SqlTypeName.ROW && outputSqlType == projectSqlType) {
+      return true;
+    }
+
+    switch (outputSqlType) {
+      case CHAR:
+        return projectSqlType == SqlTypeName.VARCHAR;
+      case VARCHAR:
+        return projectSqlType == SqlTypeName.CHAR;
+      case BIGINT:
+        return projectSqlType == SqlTypeName.INTEGER;
+      case INTEGER:
+        return projectSqlType == SqlTypeName.BIGINT;
+      case FLOAT:
+        return projectSqlType == SqlTypeName.DOUBLE;
+      case DOUBLE:
+        return projectSqlType == SqlTypeName.FLOAT;
+      case ROW:
+        try {
+          validateOutputRecords((RelRecordType) outputFieldType, (RelRecordType) projectFieldType);
+        } catch (SamzaSqlValidatorException e) {
+          LOG.error("A field in select query does not match with the output schema.", e);
+          return false;
+        }
+        return true;
+      default:
+          return false;
+    }
+  }
+
+  // -- All Static Methods below --
+
+  /**
+   * Format the Calcite exception to a more readable form.
+   *
+   * As an example, consider the below sql query which fails calcite validation due to a non existing field :
+   * "Insert into testavro.outputTopic(id) select non_existing_name, name as string_value"
+   *             + " from testavro.level1.level2.SIMPLE1 as s where s.id = 1"
+   *
+   * This function takes in the above multi-line sql query and the below sample exception as input:
+   * "org.apache.calcite.runtime.CalciteContextException: From line 1, column 8 to line 1, column 26: Column
+   * 'non_existing_name' not found in any table"
+   *
+   * And returns the following string:
+   * 2019-08-30 09:05:08 ERROR QueryPlanner:174 - Failed with exception for the following sql statement:
+   *
+   * Sql syntax error:
+   *
+   * SELECT `non_existing_name`, `name` AS `string_value`
+   * -------^^^^^^^^^^^^^^^^^^^--------------------------
+   * FROM `testavro`.`level1`.`level2`.`SIMPLE1` AS `s`
+   * WHERE `s`.`id` = 1
+   *
+   * @param query sql query
+   * @param e Exception returned by Calcite
+   * @return formatted error string
+   */
+  public static String formatErrorString(String query, Exception e) {
```

Comment:
This is going to be super useful! Thanks a lot for adding this capability :)  We just need to be careful to update this logic when When calcite is upgraded and they change the exception string.  Can we add a unit test to protect this logic when we upgrade calcite?